sdk/test-utils/recorder/test/utils.spec.ts

```diff
@@ -422,4 +427,265 @@ ultramarine.com/url/PUBLIC
+      const testAbsolutePath = path.resolve("test/myTest.spec.ts");
+
+      expect(testHasChanged(testSuiteTitle, testTitle, testAbsolutePath, newHash)).to.equal(true);
+
+      mockFs.restore();
+      mockRequire.stopAll();
+    });
+
+    it("In NodeJS - should return false if the older hash is the same as the new hash", function() {
+      if (isBrowser()) return this.skip();
+
+      const mockFs = require("mock-fs");
+      const mockRequire = require("mock-require");
+      const platform = "node";
+      const testSuiteTitle = this.test!.parent!.fullTitle();
+      const testTitle = this.test!.title;
+      const filePath = generateTestRecordingFilePath(platform, testSuiteTitle, testTitle);
+
+      // This needs to change if findRecordingsFolderPath changes.
+      mockFs({
+        // Our lazy require doesn't use the fs module internally.
+        [`recordings/${filePath}`]: "",
+        "test/myTest.spec.ts": "",
+        "../../sdk/": {},
+        "../../../rush.json": ""
+      });
+
+      mockRequire(`../recordings/${filePath}`, {
+        // We won't be testing whether MD5 works or not.
+        hash: "same old hash"
+      });
+
+      // We won't be testing whether MD5 works or not.
+      const newHash = "same old hash";
+
+      const path = require("path");
+      const testAbsolutePath = path.resolve("test/myTest.spec.ts");
+
+      expect(testHasChanged(testSuiteTitle, testTitle, testAbsolutePath, newHash)).to.equal(false);
+
+      mockFs.restore();
+      mockRequire.stopAll();
+    });
+
+    it("In NodeJS - should return true if the older hash is different than the new hash", function() {
+      if (isBrowser()) return this.skip();
+
+      const mockFs = require("mock-fs");
+      const mockRequire = require("mock-require");
+      const platform = "node";
+      const testSuiteTitle = this.test!.parent!.fullTitle();
+      const testTitle = this.test!.title;
+      const filePath = generateTestRecordingFilePath(platform, testSuiteTitle, testTitle);
+
+      // This needs to change if findRecordingsFolderPath changes.
+      mockFs({
+        // Our lazy require doesn't use the fs module internally.
+        [`recordings/${filePath}`]: "",
+        "test/myTest.spec.ts": "",
+        "../../sdk/": {},
+        "../../../rush.json": ""
+      });
+
+      mockRequire(`../recordings/${filePath}`, {
+        // We won't be testing whether MD5 works or not.
+        hash: "old hash"
+      });
+
+      // We won't be testing whether MD5 works or not.
+      const newHash = "new hash";
+
+      const path = require("path");
+      const testAbsolutePath = path.resolve("test/myTest.spec.ts");
+
+      expect(testHasChanged(testSuiteTitle, testTitle, testAbsolutePath, newHash)).to.equal(true);
+
+      mockFs.restore();
+      mockRequire.stopAll();
+    });
+
+    it("In the browser - should not crash if the recorded file doesn't exist", function() {
+      if (!isBrowser()) return this.skip();
```

Comment:
For browsers too - but in `test/browser` folder, and update rollup.test browser config accordingly.